core/unit_test/TestComplex.hpp

```diff
@@ -31,6 +30,11 @@
 #endif
 #endif
 
+namespace {
+template <typename... Ts>
+KOKKOS_FUNCTION constexpr void maybe_unused(Ts &&...) noexcept {}
+}  // namespace
```

Comment:
I would have gone all the way and move everything in this file into the anonymous namespace